components/derive/src/exonum_service.rs

```diff
@@ -151,9 +151,7 @@ impl ExonumService {
                 .iter()
                 .map(|ServiceMethodDescriptor { arg_type, id, .. }| {
                     quote! {
-                        impl #cr::runtime::rust::Transaction for #arg_type {
-                            type Service = &'static dyn #trait_name;
```

Comment:
Not sure what is the purpose of the associated type. I've changed it since in the supervisor, both the simple and advanced implementations use the same transaction type; in general, it seems counter-productive not to allow this.